### PR TITLE
SqlDataSources: Fix the Show Generated SQL button in query editors

### LINKED ISSUE #31112: QueryEditors: SQL Data sources (Postgres & MySQL) Generated SQL button does not work anymore (7.4+) due to need for panel id & wanting to open inspect drawer
Since updgrading to 7.4.0 today, the "Generated SQL" button in a Graph panel does nothing. I click it, nothing happens. Tested in both Chrome and Microsoft Edge on Windows.

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #31112

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
Looks good and works as expected. 

I noticed one difference though; compared to master, and that is that you need to refresh the query result for the `Generated SQL` button to appear.
bug

File: public/app/features/query/components/QueryEditorRow.tsx [hunk 1]
```diff
@@ -126,14 +126,16 @@ export class QueryEditorRow extends PureComponent<Props, State> {
...
+      this.setState({ data: dataFilteredByRefId });
 
       if (this.angularScope) {
         this.angularScope.range = getTimeSrv().timeRange();
       }
 
-      if (this.angularQueryEditor) {
-        notifyAngularQueryEditorsOfData(this.angularScope!, data, this.angularQueryEditor);
+      if (this.angularQueryEditor && dataFilteredByRefId) {
+        notifyAngularQueryEditorsOfData(this.angularScope!, dataFilteredByRefId, this.angularQueryEditor);
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/query/components/QueryEditorRow.tsx:138]: Don't think so, the setState is for the react query editors, and the reset is to communicate the new data to angular editors

[the remark this answers is not in the document]
Comment (hugohaggmark) [reply public/app/features/query/components/QueryEditorRow.tsx:138]: Aha of course, that's why it looks like that. Thanks for explaining.

**Comment by torkelo** [comment]
The link "Show Generated SQL" was only shown after a new query before as well, so it's sort of restoring the old behavior there.

**Pull request merged**